src/components/UserProfile/UserProfileModal/UserProfileModal.jsx

```diff
@@ -376,6 +376,30 @@ const UserProfileModal = props => {
           </Button>
         )}
 
+        {type === 'editBlueSquare' && (
+           <Button
+           color="info"
+           onClick={() => {
+             modifyBlueSquares(id, dateStamp, summary, 'update');
+           }}
+           style={boxStyle}
+         >
+           Update
+         </Button>
+        )}
+        
+        {type === 'deleteBlueSquare' && (
+          <Button
+          color="danger"
+          onClick={() => {
+            modifyBlueSquares(id, dateStamp, summary, 'delete');
+          }}
+          style={boxStyle}
+        >
+          Delete
+        </Button>
+        )}
+
```

Comment:
```suggestion
```
This is redundant. Can just add permission checks to the 'modBlueSquare' case like so.

```
{type === 'modBlueSquare' && (
          <>
            {canUpdate && 
              <Button
                color="info"
                onClick={() => {
                  modifyBlueSquares(id, dateStamp, summary, 'update');
                }}
                style={boxStyle}
              >
                Update
              </Button>
              }
            {canDelete &&
              <Button
                color="danger"
                onClick={() => {
                  modifyBlueSquares(id, dateStamp, summary, 'delete');
                }}
                style={boxStyle}
              >
                Delete
              </Button>
            }
          </>
        )}
```

Reply:
I've made the changes. Please review again